### PR TITLE
Split local and prod settings

### PR DESCRIPTION
**Pull request by kiwib0n**

### CODE REVIEW

File: cart/admin.py [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+    ColourVariation,
+    SizeVariation,
+    Address,
+    Payment,
+    Category,
+    StripePayment,
 )
 
 
+@admin.register(Address)
```
Comment (kiwib0n) [inline cart/admin.py:18]: we immediately register it through the decorator so as not to mix it with the rest

File: cart/models.py [hunk 2]
```diff
@@ -11,154 +13,192 @@ class Category(models.Model):
...
-    def __str__(self):
+    def __str__(self) -> str:
         return self.name
 
 
 class Address(models.Model):
-    ADDRESS_CHOICES = (
-        ('B', 'Billing'),
-        ('S', 'Shipping'),
+    SHIPPING_ADDRESS_TYPE: str = 'S'
```
Comment (kiwib0n) [inline cart/models.py:24]: we use constants for easy reuse

File: cart/models.py [hunk 3]
```diff
@@ -11,154 +13,192 @@ class Category(models.Model):
...
+        blank=True, null=True,
+    )
 
-    def __str__(self):
+    def __str__(self) -> str:
         return self.title
 
-    def get_absolute_url(self):
+    def save(self, *args, **kwargs):
+        if not self.slug:
```
Comment (kiwib0n) [inline cart/models.py:90]: can do without a signal

File: cart/models.py [hunk 4]
```diff
@@ -167,24 +207,32 @@ def reference_number(self):
...
+    def __str__(self) -> str:
         return self.reference_number
 
     @property
-    def reference_number(self):
+    def reference_number(self) -> str:
         return f"STRIPE-PAYMENT-{self.order}-{self.pk}"
 
 
-def pre_save_product_receiver(sender, instance, *args, **kwargs):
```
Comment (kiwib0n) [inline cart/models.py:185]: the logic of the signal is implemented by the save method of the model

File: cart/views.py [hunk 5]
```diff
@@ -61,80 +78,82 @@ def form_valid(self, form):
...
             size=form.cleaned_data['size'],
         )
-
-        if item_filter.exists():
-            item = item_filter.first()
-            item.quantity += int(form.cleaned_data['quantity'])
-            item.save()
-
-        else:
+        if not item_filter.exists():
```
Comment (kiwib0n) [inline cart/views.py:81]: can separate the business logic, create a function in actions.py for example, to facilitate the presentation

File: core/apps.py [hunk 6]
```diff
@@ -2,4 +2,10 @@
 
 
 class CoreConfig(AppConfig):
-    name = 'core'
+    name: str = 'core'
+
+    def ready(self):
+        # noinspection PyUnresolvedReferences
+        from . import signals
```
Comment (kiwib0n) [inline core/apps.py:9]: register signals when application starting

File: ecom/settings/__init__.py [hunk 7]
```diff
@@ -0,0 +1,4 @@
+try:
+    from .local import *
```
Comment (kiwib0n) [inline ecom/settings/__init__.py:2]: divide the configuration file into local and production

File: ecom/settings/local_example.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+from .main import *
```
Comment (kiwib0n) [inline ecom/settings/local_example.py:1]: an unused module that is just an example

**Comment by kiwib0n** [comment]
there is also a suggestion to add Docker support with gunicorn

**Comment by fl4viooliveira** [comment]
Thank you @kiwib0n

**Pull request merged**